### ISSUE TITLE
Does not work at all with Misa's Realistic Pack.

### ISSUE THREAD

**Issue by Nomokuto** [MehVahdJukaar/polytone#21]
Hey,
so Misa's Realistic Pack is designer for OptiFine, but I'm trying to get all features working on Fabric. The biggest issue missing at the moment is custom colors. I know it worked well with Colormatic mod, but that one isn't updated anymore so I was looking for alternatives and found Polytone. It says it offers backwards compatibility with the color.properties system previously offered by Optifine, allowing said features from existing packs to work aswell, but none of the colors semm to actually work. At least not with Misa's Realistic pack. Havn't tried other packs...
The pack can be found here: [https://www.curseforge.com/minecraft/texture-packs/misas-realistic](url) if you want to try it yourself.

**Comment by MehVahdJukaar** [comment]
That's funny because that's the very pack I used to test polytone backwards compat feature.what colours are not working?

**Comment by Nomokuto** [comment]
Oh really? The most obvious once are sand, smooth stone, gravel, small fern and water.
Might need to mention that I use Quilt and Quilted Fabric API. Havn't tried it with just Fabric yet.

**Comment by MehVahdJukaar** [comment]
Yeah try on fabric without any other mod first then same in quilt

**Comment by Nomokuto** [comment]
So just tested just with Fabric. Doesn't work.
Made a quick compare: [https://imgur.com/a/NPw3hdf](url)

**Comment by MehVahdJukaar** [comment]
i really dont like optifine formats. Polytone can do much much more than optifine yet its sad that people will never use its new features and stuff or its own formats just because that mess of optifine exist. For example optifine allows to skip defining tint index  in models when one adds a coormap is applied via color properties. polytone allows this but this stuf fjust looks super ugly and so many edge cases where it could break. for example ferns from misa just dont have tint index. as far as vanilla is concerned they cold have the color you want but just wont use it. for all im concerned they are incorrectly formatted and polytone has to resort to ungodly hacks to get that to work just because optifine set that ugly standard to begin with.
 I have however fixed what needed to be fixed so stuff will have backward compat but i still heavily discourage usage or anything related to optifine. I have a very comprehensive wiki explaining all that there is to explain when it comes to polytone formats.

**Comment by MehVahdJukaar** [comment]
oh and those ungodly hacks i use dont work on fabric because fabric has a rendering api that changes that stuff a lot. would need quite some research to find out how to do something similar there. reason it worked for me was because i tested on forge

**Comment by Nomokuto** [comment]
I don't know much. I really dislike OptiFine and the creator is also an asshole tbh. Sadly I guess it will be a ton of work for Misa to update stuff to Polytone then. But well, maybe one day she'll move to Fabric, but at the moment she seems to really dislike talking about OptiFine alternatives, even though she hates OptiFine herself..

**Comment by MehVahdJukaar** [comment]
i have found a new hack for fabric. Will only work when a block happen to have the same name as the blockstate

**Comment by MehVahdJukaar** [comment]
try new version

**Comment by Nomokuto** [comment]
Still doesn't work sadly. Now water doesn't change color at all actually heh

**Comment by MehVahdJukaar** [comment]
Water definitely works for me. It's definitely not using vanilla colours

**Comment by Nomokuto** [comment]
No, it's not working. It's the same like a in the comparing screenshot still, but the water isn't turning green in jungle with last update. Like only leaves and grass block seem to work probably. Are there any other mods, this mod need to have installed, besides Fabric and Fabric API? Cause it kinda weird, if it works on your end, but not on mine. Do you have a screenshot that it is working for you?

**Comment by MehVahdJukaar** [comment]
![Uploading image.png…]()
Yeah idk try reinstalling. the mod has no dependencies

**Comment by MehVahdJukaar** [comment]
![image](https://github.com/MehVahdJukaar/polytone/assets/24719182/ce06dee2-4877-4946-b96a-5a6959ea1c78)

**Comment by Nomokuto** [comment]
![2024-04-06_14 38 50](https://github.com/MehVahdJukaar/polytone/assets/139173187/b0dd06c1-c3c8-4d07-a6bf-de172773d19d)
Hey, so I tried this mod again cause I really want to be able to use Misa's with Fabric instead of OptiFine.
It seems like it improved. Water works now, but for me sand, gravel and fern still don't change color sadly. Using the latest version 1.17.7 on Minecrfat 1.20.1.

**Comment by MehVahdJukaar** [comment]
![image](https://github.com/MehVahdJukaar/polytone/assets/24719182/fdd63ee9-e53d-4d56-82b4-0ea8b50d2fe2)
cant replicate

**Comment by solarite1797** [comment]
> ![image](https://private-user-images.githubusercontent.com/24719182/320218204-fdd63ee9-e53d-4d56-82b4-0ea8b50d2fe2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3MzgwNTAsIm5iZiI6MTcxNjczNzc1MCwicGF0aCI6Ii8yNDcxOTE4Mi8zMjAyMTgyMDQtZmRkNjNlZTktZTUzZC00ZDU2LTgyYjQtMGVhOGI1MGQyZmUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDE1MzU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3ZDIzYWNlNmM5NGNjNTdhOWY1MGY5NTc5M2U2MGE3MGUyZGQ4ZTRlNmMyNDRhNmNkMzc4YTkyMzMxNGE5YjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ChZiGR0j5f0x7tGKClAylTmxpxdFUQ9OgCHt7hB4Ymc) cant replicate

how do people make fonts so realistic

**Comment by ItsEntDev** [comment]
> > ![image](https://private-user-images.githubusercontent.com/24719182/320218204-fdd63ee9-e53d-4d56-82b4-0ea8b50d2fe2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3MzgwNTAsIm5iZiI6MTcxNjczNzc1MCwicGF0aCI6Ii8yNDcxOTE4Mi8zMjAyMTgyMDQtZmRkNjNlZTktZTUzZC00ZDU2LTgyYjQtMGVhOGI1MGQyZmUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDE1MzU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3ZDIzYWNlNmM5NGNjNTdhOWY1MGY5NTc5M2U2MGE3MGUyZGQ4ZTRlNmMyNDRhNmNkMzc4YTkyMzMxNGE5YjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ChZiGR0j5f0x7tGKClAylTmxpxdFUQ9OgCHt7hB4Ymc) cant replicate
> 
> how do people make fonts so realistic

it isn't realistic, just a rustic font. anyway fonts are just a special kind of image format, you can do whatever you like with fonts
